src/clients/c++/perf_analyzer/client_backend/torchserve/http_client.cc

```diff
@@ -0,0 +1,347 @@
+        dynamic_cast<TorchServeInferInput*>(input);
+    this_input->PrepareForRequest();
+    bool end_of_input = false;
+    while (!end_of_input) {
+      const uint8_t* buf;
+      size_t buf_size;
+      this_input->GetNext(&buf, &buf_size, &end_of_input);
+      std::string file_path(
+          reinterpret_cast<const char*>(buf) + 4, buf_size - 4);
+      if (buf != nullptr) {
+        curl_mimepart* part = curl_mime_addpart((curl_mime*)mime_handle_);
+        curl_mime_filedata(part, file_path.c_str());
+        curl_mime_name(part, "data");
+        if (verbose_) {
+          input_filepaths.push_back(file_path);
+        }
+      }
+    }
+  }
+  curl_easy_setopt(easy_handle_, CURLOPT_MIMEPOST, (curl_mime*)mime_handle_);
+
+  // response headers handled by InferResponseHeaderHandler()
+  curl_easy_setopt(curl, CURLOPT_HEADERFUNCTION, InferResponseHeaderHandler);
+  curl_easy_setopt(curl, CURLOPT_HEADERDATA, http_request.get());
+
+  // response data handled by InferResponseHandler()
+  curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, InferResponseHandler);
+  curl_easy_setopt(curl, CURLOPT_WRITEDATA, http_request.get());
+
+  struct curl_slist* list = nullptr;
+  for (const auto& pr : headers) {
+    std::string hdr = pr.first + ": " + pr.second;
+    list = curl_slist_append(list, hdr.c_str());
+  }
+  curl_easy_setopt(curl, CURLOPT_HTTPHEADER, list);
+
+  // The list will be freed when the request is destructed
+  http_request->header_list_ = list;
+
+  if (verbose_) {
+    std::cout << "inference request : [";
+    bool first = true;
+    for (const auto& fn : input_filepaths) {
+      if (first) {
+        first = false;
+      } else {
+        std::cout << ",";
+      }
+      std::cout << "\"" << fn << "\"";
+    }
+    std::cout << "]" << std::endl;
+  }
+
+  return Error::Success;
+}
+
+HttpClient::HttpClient(const std::string& url, bool verbose)
+    : InferenceServerClient(verbose), url_(url),
+      easy_handle_(reinterpret_cast<void*>(curl_easy_init()))
+{
+}
+
+HttpClient::~HttpClient()
+{
+  exiting_ = true;
+
+  if (easy_handle_ != nullptr) {
+    curl_easy_cleanup(reinterpret_cast<CURL*>(easy_handle_));
+  }
+}
+
+//======================================================================
+Error
+InferResult::Create(
+    InferResult** infer_result, std::shared_ptr<HttpInferRequest> infer_request)
+{
+  *infer_result =
+      reinterpret_cast<InferResult*>(new InferResult(infer_request));
+  return Error::Success;
+}
+
+Error
+InferResult::RequestStatus() const
+{
+  return status_;
+}
+
+InferResult::InferResult(std::shared_ptr<HttpInferRequest> infer_request)
+    : infer_request_(infer_request)
+{
+  if (infer_request->http_code_ != 200) {
+    status_ = Error(
+        "inference failed with error code " +
+        std::to_string(infer_request->http_code_));
+  }
+}
+
+//======================================================================
+
+}}}  // namespace perfanalyzer::clientbackend::torchserve
```

Comment:
Add newline

Reply:
You mean EOF line right? It is already there. github does not render it by default.